src/Features/Core/Portable/RemoveUnnecessaryParentheses/AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer.cs

```diff
@@ -126,8 +129,16 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
 
             var additionalLocations = ImmutableArray.Create(parenthesizedExpression.GetLocation());
 
-            // Fades the open parentheses character and reports the suggestion.
-            context.ReportDiagnostic(Diagnostic.Create(s_diagnosticWithFade, parenthesizedExpression.GetFirstToken().GetLocation(), additionalLocations));
+            // Since this diagnostic descriptor is not user configurable, do not report it if this diagnostic id is hidden / suppressed.
```

Comment:
NotConfigurable means end user can never change the effective severity of the reported diagnostic - it is always reported at the default severity from the descriptor. The fact that the user is able to change the severity of your fading descriptor indicates that both the fading and non-fading diagnostics are still configurable by user, and your change here is filtering out a specific configuration case.

I think a better fix is to retain the original code in this method and just switch the fading descriptor to use a different diagnostic ID and make it non-configurable.

Reply:
@mavasani That will definitely work for the error case, but if the user sets the diagnostic to "none" in ruleset it will still fade the parentheses.  If that is fine then I can just do not configurable, the reason I added the effective severity was so that it wouldn't be faded in that case.
I see Sam mentioned hidden should be faded, but what about none/suppressed?